extensions/data-plane/data-plane-aws-s3/src/main/java/org/eclipse/edc/connector/dataplane/aws/s3/S3DataSink.java

```diff
@@ -89,11 +93,19 @@ protected StreamResult<Object> transferParts(List<DataSource.Part> parts) {
         return StreamResult.success();
     }
 
-    private String getDestinationObjectName(String partName) {
+    private String getDestinationObjectName(String partName, int partsSize) {
+        if (isNullOrEmpty(objectName) && !isNullOrEmpty(keyName)) {
+            monitor.warning("The usage of the property \"keyName\" to define the object name is deprecated" +
```

Comment:
the destination is chosen by the consumer and this code gets run on the provider, so there's no point in logging this information on the provider side as it has no control on it, or, better, it has, through the destination data address validator (transfer won't happen if there's no objectName).

Reply:
Yes, There was no reason for that, I completely removed the check as the new attributes are also optional.